test/spmd/test_inference_spmd_dynamo_imagenet.py

```diff
@@ -0,0 +1,238 @@
+        batch_size=FLAGS.test_set_batch_size,
+        sampler=None,
+        drop_last=FLAGS.drop_last,
+        shuffle=False)
+
+  torch.manual_seed(42)
+
+  model = torchvision.models.resnet50().to(
+      device)  # get_model_property('model_fn')().to(device)
+
+  input_mesh = None
+  if FLAGS.sharding:
+    num_devices = pjrt.global_device_count()
+    device_ids = np.arange(num_devices)
+    # Model sharding
+    if 'conv' in FLAGS.sharding:
+      # Shard the model's convlution layers along two dimensions
+      mesh_shape = (2, num_devices // 2, 1, 1)
+      mesh = xs.Mesh(device_ids, mesh_shape, ('w', 'x', 'y', 'z'))
+      partition_spec = (0, 1, 2, 3)  # Apply sharding along all axes
+      print(
+          f'Applying sharding to convolution layers with mesh {mesh.get_logical_mesh()}'
+      )
+      for name, layer in model.named_modules():
+        if 'conv' in name:
+          xs.mark_sharding(layer.weight, mesh, partition_spec)
+    elif 'linear' in FLAGS.sharding:
+      # Shard the model's fully connected layers across addressable devices
+      mesh_shape = (num_devices, 1)
+      mesh = xs.Mesh(device_ids, mesh_shape, ('x', 'y'))
+      print(
+          f'Applying sharding to linear layers with mesh {mesh.get_logical_mesh()}'
+      )
+      partition_spec = (0, 1)
+      for name, layer in model.named_modules():
+        if 'fc' in name:
+          xs.mark_sharding(layer.weight, mesh, partition_spec)
+
+    # Input sharding
+    if 'batch' in FLAGS.sharding or 'spatial' in FLAGS.sharding:
+      if 'batch' in FLAGS.sharding and 'spatial' in FLAGS.sharding:
+        # Shard along both the batch dimension and spatial dimension
+        # If there are more than 4 devices, shard along the height axis as well
+        width_axis, height_axis = 2, num_devices // 4
+        mesh_shape = (2, 1, width_axis, height_axis)
+        input_mesh = xs.Mesh(device_ids, mesh_shape, ('B', 'C', 'W', 'H'))
+        print(
+            f'Sharding input along batch and spatial dimensions with mesh {input_mesh.get_logical_mesh()}'
+        )
+      elif 'batch' in FLAGS.sharding:
+        # Shard along batch dimension only
+        mesh_shape = (num_devices, 1, 1, 1)
+        input_mesh = xs.Mesh(device_ids, mesh_shape, ('B', 'C', 'W', 'H'))
+        print(
+            f'Sharding input along batch dimension with mesh {input_mesh.get_logical_mesh()}'
+        )
+      elif 'spatial' in FLAGS.sharding:
+        # Shard two-way along input spatial dimensions
+        mesh_shape = (1, 1, num_devices // 2, 2)
+        input_mesh = xs.Mesh(device_ids, mesh_shape, ('B', 'C', 'W', 'H'))
+        print(
+            f'Sharding input images on spatial dimensions with mesh {input_mesh.get_logical_mesh()}'
+        )
+      test_loader = pl.MpDeviceLoader(
+          test_loader,
+          device,
+          input_sharding=xs.ShardingSpec(input_mesh, (0, 1, 2, 3)))
+
+  model.eval()
+  xm.mark_step()
+  xm.wait_device_ops()
+  met.clear_all()
```

Comment:
why are these needed? I had these in my dynamo test because test data is fake so I need a `mark_step` to materalize the fake input, if that;'s not the case here we don't need to do a mark_step here.

Reply:
We need to materialize the input here as well. @yeounoh may be able to add more context

Reply:
ok, maybe add a comment above to mentioned that this `mark_step` is to materialize the fake input then.

Reply:
Done